ring.go

```diff
@@ -18,6 +18,24 @@ func (r *Ring) Size() int {
 	return len(r.pubkeys)
 }
 
+// Equals checks whether the supplied ring is equal to the current ring.
```

Comment:
I'd add a comment says "the public keys must be in the same order for the rings to be equal" or something along those lines